### ISSUE TITLE
Solver enters infinite loop with quadprog==0.1.10

### ISSUE THREAD

**Issue by stephane-caron** [quadprog/quadprog#32]
@junhyeokahn reported in https://github.com/stephane-caron/qpsolvers/issues/47 that quadprog enters an infinite loop on the following QP: [qp.zip](https://github.com/stephane-caron/qpsolvers/files/7508138/qp.zip)

Here is the same ``main.py`` adapted to use quadprog directly:

```python
import numpy as np
from quadprog import solve_qp

def quadprog_solve_qp(P, q, G, h, A, b):
    qp_G, qp_a, qp_C, qp_b = P, -q, -np.vstack([A, G]).T, -np.hstack([b, h])
    return solve_qp(qp_G, qp_a, qp_C, qp_b, A.shape[0])[0]

cost_mat = np.loadtxt("cost_mat.txt")
cost_vec = np.loadtxt("cost_vec.txt")
eq_mat = np.loadtxt("eq_mat.txt")
eq_vec = np.loadtxt("eq_vec.txt")
ineq_mat = np.loadtxt("ineq_mat.txt")
ineq_vec = np.loadtxt("ineq_vec.txt")

sol = quadprog_solve_qp(cost_mat, cost_vec, ineq_mat, ineq_vec, eq_mat, eq_vec)
print(sol)
```

On my machine both quadprog 0.1.8 and 0.1.9 solve this problem but quadprog 0.1.10 indeed enters an infinite loop (use Ctrl-\ to stop the script as it doesn't respond to Ctrl-C).

**Comment by LiZhongyangLi** [comment]
I find it is common to run into this for three variables but much less common for two. Here is another example:

```
import numpy as np
import quadprog

G = np.array([[224.60560028, 181.38561347, 299.23703769],
              [181.38561347, 148.05984179, 238.58321617],
              [299.23703769, 238.58321617, 406.34542188]])
a= np.array([239.91135277, 196.08680183, 313.40206452])
C = np.array([[-1.,  1.,  1.,  0.,  0.],
              [-1.,  1.,  0.,  1.,  0.],
              [-1.,  1.,  0.,  0.,  1.]])
b = np.array([-1.19, 0.7, 0., 0., 0.])
meq = 0
quadprog.solve_qp(G, a, C, b, meq)
```

**Comment by preeyan** [comment]
Thank you @junhyeokahn, @stephane-caron and @LiZhongyangLi for investigating.

This was a bug accidentally introduced when fixing #2 and unfortunately not covered by the existing test cases.

I have added @LiZhongyangLi test case to the unit tests, fixed the bug and released 0.1.11.
